r/geoarrow/src/r-vctr.c

```diff
@@ -0,0 +1,116 @@
+
+#define R_NO_REMAP
+#include <R.h>
+#include <Rinternals.h>
+
+#include "geoarrow.h"
+
+SEXP geoarrow_c_vctr_chunk_offsets(SEXP array_list) {
```

Comment:
Do we expect `array_list` to be large? If not, should this logic be moved to R? The overhead should be tiny if `array_list` is small.

```r
chunk_lengths <- vapply(chunks, function(x) x$length, integer(1L))
c(0L, cumsum(chunk_lengths))
```

Reply:
I'm definitely not thinking of 100 million length geometry vectors!